src/formatDate.js

```diff
@@ -8,7 +8,38 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const fromSeparator = fromFormat.pop();
+  const toSeparator = toFormat.pop();
+  const dateParts = date.split(fromSeparator);
+  const objFrom = {};
+  const arrTo = [];
+
+  for (let i = 0; i < fromFormat.length; i++) {
+    objFrom[fromFormat[i]] = dateParts[i];
+  }
+
+  for (let i = 0; i < toFormat.length; i++) {
+    for (const entry of Object.entries(objFrom)) {
+      const twoLettersTo = toFormat[i].slice(0, 2);
+      const twoLettersFrom = entry[0].slice(0, 2);
+
+      if (toFormat[i] === entry[0]) {
+        arrTo[i] = entry[1];
+      } else if (twoLettersTo === twoLettersFrom) {
+        if (entry[0].length > toFormat[i].length) {
+          arrTo[i] = entry[1].slice(2, 4);
+        } else {
+          if (+entry[1] < 30) {
+            arrTo[i] = '20' + entry[1];
+          } else {
+            arrTo[i] = '19' + entry[1];
+          }
+        }
+      }
```

Comment:
This nested loop with Object.entries is inefficient as it will iterate through all entries for every element in `toFormat`. Consider restructuring the algorithm to avoid unnecessary iterations. Additionally, there is no check to ensure that `objFrom` contains the expected parts of the date, which might cause issues if the input date format is incorrect.

Reply:
Consider refactoring your code. You can use 2 separate loops instead of nested